From 5f4ed9e59e8912c5d304be7e920ad5b34066319f Mon Sep 17 00:00:00 2001
From: Aniket Limaye <a-limaye@ti.com>
Date: Tue, 2 Aug 2022 00:20:48 +0530
Subject: [PATCH] arm64: dts: ti: rtos-memory-map: Add reserved-memory-node
 for CMA of size 512MB in upper 2GB

Not enough room to reserve 512MB of CMA memory in lower 2GB

Possible alternate fix: move C7x memory region to upper 2GB to make space for CMA
---
 arch/arm64/boot/dts/ti/k3-j721s2-rtos-memory-map.dtsi | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/arch/arm64/boot/dts/ti/k3-j721s2-rtos-memory-map.dtsi b/arch/arm64/boot/dts/ti/k3-j721s2-rtos-memory-map.dtsi
index fd81d387b..6d8a5416f 100644
--- a/arch/arm64/boot/dts/ti/k3-j721s2-rtos-memory-map.dtsi
+++ b/arch/arm64/boot/dts/ti/k3-j721s2-rtos-memory-map.dtsi
@@ -199,6 +199,12 @@
 		reg = <0x08 0x80000000 0x00 0x10000000>;
 		no-map;
 	};
+        linux_cma_region: linux-cma-buffers@890000000 {
+                compatible = "shared-dma-pool";
+                reusable;
+                reg = <0x08 0x90000000 0x00 0x20000000>;
+                linux,cma-default;
+        };
 };
 
 &mcu_r5fss0_core0 {
-- 
2.17.1

